frontend/tests/emailWorkflow.test.ts

```diff
@@ -152,3 +153,107 @@ test('register via new-user invitation email', async ({ page, mailboxFactory })
+    let dashboardPage = await new UserDashboardPage(page).goto();
+
+    // Must verify email before being made manager of a project
+    await dashboardPage.emailVerificationAlert.assertPleaseVerify();
+    let newPage = await verifyTempUserEmail(page, manager);
+
+    // Add manager to Elawa project
+    await loginAs(newPage.request, 'admin');
+    await addUserToProject(newPage.request, manager.id, elawaProjectId, 'MANAGER');
+
+    const { name, email, password } = tempUserInTestOrg;
+
+    await loginAs(page.request, email, password);
+    dashboardPage = await new UserDashboardPage(page).goto();
+
+    // Must verify email before being allowed to request project creation
+    await dashboardPage.emailVerificationAlert.assertPleaseVerify();
+    newPage = await verifyTempUserEmail(page, tempUserInTestOrg);
+    dashboardPage = await new UserDashboardPage(newPage).goto();
+
+    // Create project with similar name to Elawa
+    const newProjectPage = await dashboardPage.clickCreateProject();
+    await newProjectPage.fillForm({name: 'Elaw', code: 'xyz', purpose: 'Testing', organization: 'Test Org'});
+    await expect(newProjectPage.extraProjectsDiv).toBeVisible();
+    await expect(newProjectPage.askToJoinBtn).toBeDisabled();
+    await newProjectPage.extraProjectsDiv.locator('#extra-projects-elawa-dev-flex').check();
+    await expect(newProjectPage.askToJoinBtn).toBeEnabled();
+    await newProjectPage.askToJoinBtn.click();
+    await expect(newProjectPage.toast('has been sent to the project manager(s)')).toBeVisible();
+
+    // Log in as manager, approve join request.
+    await loginAs(page.request, manager.email, manager.password);
+    const emailPage = await manager.mailbox.openEmail(page, `${EmailSubjects.ProjectJoinRequest}: ${name}`);
+    const pagePromise = emailPage.page.context().waitForEvent('page');
+    await emailPage.clickApproveRequest();
+    newPage = await pagePromise;
+    const elawaProjectPage = await new ProjectPage(newPage, 'Elawa', 'elawa-dev-flex').waitFor();
+    await elawaProjectPage.modal.getByRole('button', {name: 'Add Member'}).click();
+    await elawaProjectPage.goto();
+
+    // Log in as temp user, should see Elawa project
+    await loginAs(page.request, email, password);
+    dashboardPage = await new UserDashboardPage(page).goto();
+    await dashboardPage.openProject('Elawa', 'elawa-dev-flex');
+});
+
+test('ask to join project via project page', async ({ page, tempUser, tempUserInTestOrg }) => {
+    test.setTimeout(TEST_TIMEOUT_2X);
+
+    // First, set up new user to be manager of Elawa project (since it doesn't have one in default seed data)
+    const manager = tempUser;
+    await loginAs(page.request, manager.email, manager.password);
+    let dashboardPage = await new UserDashboardPage(page).goto();
+
+    // Must verify email before being made manager of a project
+    await dashboardPage.emailVerificationAlert.assertPleaseVerify();
```

Comment:
just a thought, since we're not actually testing the verification we could skip this. And since email verification does not require login we could actually skip logging in, going to the dashboard and checking for verification. The same can be done for the user and in the other test as well

Reply:
Done in commit 531bd7c5.